consumer/consumererror/error.go

```diff
@@ -0,0 +1,145 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+	"google.golang.org/grpc/status"
+)
+
+type Error struct {
+	errors []ErrorData
+}
+
+type ErrorData interface {
+	Error() string
+
+	Unwrap() error
+
+	// Returns nil if a status is not available.
+	HTTPStatus() (int, bool)
+
+	// Returns nil if a status is not available.
+	GRPCStatus() (*status.Status, bool)
+
+	Metadata() []any
```

Comment:
This will serve as the extension point for these errors. Users who want custom data propagated back up the pipeline can do so here.

Reply:
let's maybe remove it for now and make this interface not implementable so we can easily extend it in the future?

Reply:
I'm okay with that, I've removed it. Also meant to restrict interface implementation, did that as well.